sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -45,11 +45,18 @@ private[hive] class SparkExecuteStatementOperation(
     parentSession: HiveSession,
     statement: String,
     confOverlay: JMap[String, String],
-    runInBackground: Boolean = true)
+    runInBackground: Boolean = true,
+    queryTimeout: Long)
   extends ExecuteStatementOperation(parentSession, statement, confOverlay, runInBackground)
   with SparkOperation
   with Logging {
 
+  private val queryTimeoutValue = if (queryTimeout <= 0) {
+    sqlContext.conf.getConf(SQLConf.THRIFTSERVER_QUERY_TIMEOUT)
+  } else {
+    queryTimeout
+  }
+
```

Comment:
I don't understand this. If the passed queryTimeout value is <= 0 why would we enforce the server-side configuration? And what is the relationship with setting `java.sql.Statement.setQueryTimeout`?

Reply:
This code assumes the server-side config is a global timeout value. If a user sets a timeout value via `setQueryTimeout`, the server uses the user-specified value instead of global one.